azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/repository/storage/MessageRepositoryTreeStorageUnitTest.kt

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.chat.repository.storage
+
+import com.azure.android.communication.ui.chat.models.MessageInfoModel
+import com.azure.android.communication.ui.chat.repository.MessageRepository
+import com.azure.android.communication.ui.chat.service.sdk.wrapper.ChatMessageType
+import org.junit.Assert
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class MessageRepositoryTreeStorageUnitTest {
```

Comment:
We can consolidate MessageRepository tests to cover multiple Storage implementations with one set of tests. Doesn't have to be this PR

Reply:
I needed to validate if tree based implementations were giving correct output for APIs. 

In my opinion, since the underlaying implementations are separate for each type of implementation. Its safer to have them tested separately rather relying on messageRepo tests. Otherwise messageRepo tests will keep growing and will be hard to follow the coverage of underlaying implementation.

Reply:
If the Message Repo API/Contract is doing what it should, the implementation is hidden, that is what we want.

The methods that need testing are still just the things from the reader/writer interfaces, which are exposed via delegates on the message repo.

So a very thorough test suite of MessageRepo is what we want, with all tests passing when running the tests with the chosen storage implementation. That way if we make a new Storage Implementation, you don't need a new test suite, you just tell it "use this implementation" instead and can do TDD to finish it.

Reply:
I'll resolve for now though, don't want it to hold up this pr